hub/util/shape.py

```diff
@@ -47,20 +47,24 @@ def __init__(self, lower: Sequence[int], upper: Sequence[int] = None):
         self._lower = tuple(lower)
         self._upper = tuple(upper)
 
+    def astuple(self) -> Tuple[Optional[int], ...]:
+        # TODO: named tuple? NHWC shape would be (10, 224, 224, 3) could be (N=10, H=224, W=224, C=3).
```

Comment:
separate task required, these `field_names` would need to be determined in each corresponding `htype`.